internal/pkg/cli/secret_init.go

```diff
@@ -146,33 +170,44 @@ func (o *secretInitOpts) Ask() error {
 	return nil
 }
 
-// Execute creates the secrets.
+// Execute creates or updates the secrets.
 func (o *secretInitOpts) Execute() error {
-	secretName := o.name
-	secretValues := o.values
-
 	if o.inputFilePath != "" {
-		var err error
-		secretName, secretValues, err = o.parseFile()
+		secrets, err := o.parseSecretsInputFile()
 		if err != nil {
 			return err
 		}
+		for secretName, secretValues := range secrets {
+			o.putSecret(secretName, secretValues)
+		}
+		return nil
 	}
 
-	return o.putSecrets(secretName, secretValues)
+	o.putSecret(o.name, o.values)
+	return nil
 }
 
-func (o *secretInitOpts) putSecrets(secretName string, values map[string]string) error {
+func (o *secretInitOpts) putSecret(secretName string, values map[string]string) {
+	spinner := termprogress.NewSpinner(log.DiagnosticWriter)
+
+	spinner.Start("start creating secret")
+	errLogs := make([]string, 0)
 	for envName, value := range values {
-		err := o.putSecret(secretName, envName, value)
+		err := o.putSecretInEnv(secretName, envName, value)
 		if err != nil {
-			return err
+			errLogs = append(errLogs, fmt.Sprintf("Failed to put secret %s in environment %s: %s", color.HighlightUserInput(secretName), color.HighlightUserInput(envName), err.Error()))
```

Comment:
can we log the error earlier [here](https://github.com/aws/copilot-cli/pull/2301/files#diff-f89e17f1e021d27071747e80e78111d9fb921a1d3af52a0346b0de25baef4315R238)? so that it would be more consistent. what do you think?

Reply:
I did this because the error log is usually a larger block since it is appended by errors returned from the call stack.

This makes the success/warning messages rather unnoticeable as they may be squished between larger blocks of error logs.

What consistency are you concerned with?

Reply:
ok i think i got it.